File: src/mantle/blocks/iface/IFacingLogic.scala
package mantle.blocks.iface

import net.minecraftforge.common.ForgeDirection
import net.minecraft.entity.EntityLivingBase

/**
 * Logic for blocks which have a specific direction facing.
 *
 * @author mDiyo
 * @author Sunstrike
 */
trait IFacingLogic {

  def getRenderDirection: Byte

  def getForgeDirection: ForgeDirection

  @Deprecated
  def setDirection(side: Int)

  @Deprecated
  def setDirection(yaw: Float, pitch: Float, player: EntityLivingBase)

}

File: src/mantle/blocks/iface/IFacingLogic.java
package mantle.blocks.iface;

import net.minecraft.entity.EntityLivingBase;
import net.minecraft.util.EnumFacing;

/**
 * Logic for blocks which have a specific direction facing.
 *
 * @author mDiyo
 */
public interface IFacingLogic
{

    public byte getRenderDirection();

    public EnumFacing getForgeDirection();

    public void setDirection(EnumFacing facing, float yaw, float pitch, EntityLivingBase player);

}
